Sources/muterCore/TestReporting/HTML/HTMLReporter.swift

```diff
@@ -0,0 +1,264 @@
+            <tr>
+              <td class="left-aligned">\(report.fileName)</td>
+              <td class="right-aligned">\(report.appliedOperators.count)</td>
+              <td class="right-aligned" style="color: \(report.scoreColor);">\(report.mutationScore)</td>
+            </tr>
+            """
+        }.joined(separator: "\n")
+
+        return """
+        <table id=\"mutation-operators-per-file\">
+            \(thead)
+            <tbody>
+            \(tbody)
+            </tbody>
+        </table>
+        """
+    }
+
+    func operatorsHTMLTable() -> String {
+        let thead =
+            """
+            <thead>
+                <tr>
+                    <th>File</th>
+                    <th>Applied Mutation Operator</th>
+                    <th>Changes</th>
+                    <th>Mutation Test Result</th>
+                </tr>
+            </thead>
+            """
+
+        let tbody = sorted().compactMap { report in
+            report.appliedOperators.compactMap { appliedOperator in
+                """
+                <tr>
+                    <td class="left-aligned">\(report.fileName):\(appliedOperator.mutationPoint.position.line)</td>
+                    <td class="left-aligned"><wbr>\(appliedOperator.mutationPoint.mutationOperatorId.friendlyName)<wbr></td>
+                    <td class="mutation-snapshot">\(appliedOperator.diff)</td>
+                    <td>\(appliedOperator.testSuiteOutcome.asIcon)</td>
+                </tr>
+                """
+            }.joined(separator: "\n")
+        }.joined(separator: "\n")
+
+        return """
+        <table id=\"applied-operators\">
+            \(thead)
+            <tbody>
+            \(tbody)
+            </tbody>
+        </table>
+        """
+    }
+}
+
+private extension MutationOperator.Id {
+    var friendlyName: String {
+        switch self {
+        case .ror: return "Relational Operator Replacement"
+        case .removeSideEffects: return "Remove Side Effects"
+        case .logicalOperator: return "Change Logical Connector"
+        }
+    }
+}
+
+private extension MuterTestReport.AppliedMutationOperator {
+    var diff: String {
+        var diff = ""
+        if mutationPoint.mutationOperatorId == .removeSideEffects {
+            diff = """
+            <span class="snapshot-before">\(mutationSnapshot.before)</span>
+            """
+        } else {
+            diff = """
+            <span class="snapshot-before">\(mutationSnapshot.before)</span>
+            <span class="snapshot-arrow">→</span>
+            <span class="snapshot-after">\(mutationSnapshot.after)</span>
+            """
+        }
+
+        return """
+        <div class="snapshot-changes">
+            \(diff)
+        </div>
+        """
+    }
+}
+
+private extension TestSuiteOutcome {
+    var asIcon: String {
+        switch (self) {
+        case .passed:
+            return """
+                    <svg class="failed" role="img" viewBox="0 0 72 72">
```

Comment:
It would be nice to have the SVGs as resources so we can see and edit them later. They could still be embedded in the page, but having them as files would be good for maintainability.

Reply:
good call!
I'll also do it for the other resources like css and javascript!